torchrl/envs/transforms/transforms.py

```diff
@@ -251,15 +256,19 @@ def parent(self) -> EnvBase:
                 raise ValueError(
                     f"Compose parent was of type {type(compose_parent)} but expected TransformedEnv."
                 )
+            if compose_parent.transform is not compose:
+                comp_parent_trans = compose_parent.transform
+                comp_parent_trans.reset_parent()
```

Comment:
are we sure that the whole transformed env remains unchanged after we call `transform.parent` in this case?